test/box-luatest/gh_9979_tuple_format_validate_test.lua

```diff
@@ -0,0 +1,119 @@
+            {'key5', box.NULL, 0, {}, {status = 'none'}},
+            {'k1', 'v1', 999999, {}, {size = 'big'}},
+            {'key6', {nested = {more = {levels = true}}}, 1, {'tag'},
+                {structure = 'ok'}},
+            {'t1', 'b1', 123, {'x', 'y', 'z'}, {list = {1, 2, 3}, note = 'yes'}},
+            {'json1', {x = 1}, 10, {'json'}, {data = {valid = true}}},
+        }
+
+        for _, case in ipairs(valid_cases) do
+            format:validate(case)
+            local tuple_case = box.tuple.new(case)
+            format:validate(tuple_case)
+        end
+
+        local invalid_cases = {
+            {
+                -- key not a string
+                value = {1, 'value', 1, {}, {}},
+                field = 1, name = 'key', expected = 'string',
+                actual = 'unsigned',
+            },
+            {
+                -- count negative
+                value = {'key', 'value', -1, {}, {}},
+                field = 3, name = 'count', expected = 'unsigned',
+                actual = 'integer',
+            },
+            {
+                -- metadata not a map
+                value = {'key', 'value', 1, {}, {}},
+                field = 5, name = 'metadata', expected = 'map',
+                actual = 'array',
+            },
+            {
+                -- tags not an array
+                value = {'key', 'value', 1, 'not_array', {}},
+                field = 4, name = 'tags',
+                expected = 'array', actual = 'string',
+            },
+        }
+
+        for _, case in ipairs(invalid_cases) do
+            local message = string.format(
+                "Tuple field %d (%s) type does not match one required " ..
+                "by operation: expected %s, got %s",
+                case.field, case.name, case.expected, case.actual
+            )
+
+            t.assert_error_covers({
+                type = 'ClientError',
+                code = box.error.FIELD_TYPE,
+                message = message,
+            },
+            format.validate, format, case.value)
+
+            local tuple_case = box.tuple.new(case.value)
+            t.assert_error_covers({
+                type = 'ClientError',
+                code = box.error.FIELD_TYPE,
+                message = message,
+            },
+            format.validate, format, tuple_case)
+        end
+    end)
+end
+
+g.test_format_validate_usage_error = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local format = box.tuple.format.new({{name = 'a', type = 'unsigned'}})
+
+        -- no args
+        t.assert_error_msg_equals(
+            'Usage: format:validate(tuple_or_table)',
+            format.validate, format
+        )
+
+        -- wrong arg types
+        for _, bad in ipairs({123, 'str', true, box.NULL}) do
+            t.assert_error_msg_equals(
+                'Expected tuple or table',
+                format.validate, format, bad
```

Comment:
Let's also add a case when we pass an invalid object instead of the `format` itself.